TestGoMinorVersion fails on Go RCs

Fedora 32 is going to have Go 1.14, but since neither are released yet, it's on the release candidate. This causes a failure in `TestGoMinorVersion` because the minor is incorrectly parsed:

```
--- FAIL: TestGoMinorVersion (0.00s)
    quicktest.go:306: 
        error:
          values are not equal
        got:
          bool(false)
        want:
          bool(true)
        stack:
          /builddir/build/BUILD/hugo-0.64.1/_build/src/github.com/gohugoio/hugo/common/hugo/version_test.go:87
            c.Assert(GoMinorVersion() >= 11, qt.Equals, true)
```

The Go runtime version is "1.14rc1".

### What version of Hugo are you using (`hugo version`)?

<pre>
$ hugo version
0.64.1
</pre>

### Does this issue reproduce with the latest release?

Yes

--- a/common/hugo/version.go
+++ b/common/hugo/version.go
@@ -15,7 +15,7 @@ package hugo
 
 import (
 	"fmt"
-	"strconv"
+	"io"
 
 	"runtime"
 	"strings"
@@ -245,7 +245,15 @@ func goMinorVersion(version string) int {
 	if strings.HasPrefix(version, "devel") {
 		return 9999 // magic
 	}
-	i, _ := strconv.Atoi(strings.Split(version, ".")[1])
-	return i
-
+	var major, minor int
+	var trailing string
+	n, err := fmt.Sscanf(version, "go%d.%d%s", &major, &minor, &trailing)
+	if n == 2 && err == io.EOF {
+		// Means there were no trailing characters (i.e., not an alpha/beta)
+		err = nil
+	}
+	if err != nil {
+		return 0
+	}
+	return minor
 }

--- a/common/hugo/version_test.go
+++ b/common/hugo/version_test.go
@@ -84,5 +84,6 @@ func TestParseHugoVersion(t *testing.T) {
 func TestGoMinorVersion(t *testing.T) {
 	c := qt.New(t)
 	c.Assert(goMinorVersion("go1.12.5"), qt.Equals, 12)
+	c.Assert(goMinorVersion("go1.14rc1"), qt.Equals, 14)
 	c.Assert(GoMinorVersion() >= 11, qt.Equals, true)
 }

